src/shmem_team.c

```diff
@@ -334,11 +334,29 @@ int shmem_internal_team_split_strided(shmem_internal_team_t *parent_team, int PE
         myteam->start       = global_PE_start;
         myteam->stride      = PE_stride;
         myteam->size        = PE_size;
-        if (config) {
-            myteam->config      = *config;
-            myteam->config_mask = config_mask;
+
+        if (config_mask == 0) {
+            shmem_team_config_t defaults;
+            myteam->config_mask   = 0;
+            myteam->contexts_len  = 0;
+            defaults.num_contexts = 0;
+            memcpy(&myteam->config, &defaults, sizeof(shmem_team_config_t));
```

Comment:
Do we have to assert that `config` argument is NULL/empty in this case?

Reply:
If we use assert(), my understanding is that's only built for debug.  Seems we need a NULL check for release?

Reply:
The spec says a `config_mask` value of 0 indicates that all `config` parameters should be set to the default values, so I don't think we need to check `config`.  However, it's likely beneficial to users to know the config is being ignored in case they accidentally pass a 0 mask.  Perhaps we can add a check if `ENABLE_ERROR_CHECKING` is defined?

Reply:
Yes, that sounds good. Keeping it under `ENABLE_ERROR_CHECKING` also makes sense.

Reply:
@wrrobin/@bcmIntc - I implemented the error check, but now I'm reconsidering, because this case probably shouldn't generate an error (e.g., the test in https://github.com/openshmem-org/tests-sos/pull/37 is valid usage, but would fail with error-checking).  How about printing a warning instead?

Reply:
@davidozog - Oh.. sorry, I missed that you meant as an error. I was actually thinking only a warning to the developer that the supplied config cannot be used and ignored.

Reply:
@wrrobin, currently `ENABLE_ERROR_CHECKING` only reports errors (resulting in aborts), and the checks are mostly on the critical path (so best to enable by default).  I think a simple warning is better in this case... and it's not on the critical path so we can just keep the warning around.  Do you agree?

Reply:
Yes. Keeping it as debug message is another option, but I think warning is better.